src/backend/distributed/sql/udfs/citus_shards/latest.sql

```diff
@@ -27,6 +22,10 @@ JOIN
    pg_dist_partition
 ON
    pg_dist_partition.logicalrelid = pg_dist_shard.logicalrelid
+LEFT JOIN
+   (SELECT (regexp_matches(table_name,'_(\d+)$'))[1]::int as shard_id, max(size) as size from citus_shard_sizes() GROUP BY shard_id) as shard_sizes
```

Comment:
It would be nice if `citus_shard_sizes()` returned the shard_id too, then we wouldn't need this regex. But that would require some more changes and this regex works fine too. 

The GROUP BY and `max` are to ensure we only get one size per shard in case of rebalancing going on in the mean time.